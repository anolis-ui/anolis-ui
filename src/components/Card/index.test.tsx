import "jest-styled-components";

import { Card } from ".";
import { x } from "@xstyled/styled-components";
import { FC } from "react";
import renderer from "react-test-renderer";

test("Card component", () => {
  const component = renderer.create(
    <>
      <Card
        title="Card title"
        media={<x.img src="http://placekitten.com/500/300" alt="cat" />}
        footer={<x.p>Footer</x.p>}
      >
        <LoremIpsum units="sentences" />
      </Card>

      <Card
        title="Card title"
        media={<x.img src="http://placekitten.com/500/300" alt="cat" />}
        body={<LoremIpsum units="sentences" />}
        footer={<x.p>Footer</x.p>}
      />

      <Card
        title="Card title"
        tag="Tag"
        media={<x.img src="http://placekitten.com/500/300" alt="cat" />}
        body={<LoremIpsum units="sentences" />}
      />

      <Card
        title="Card title"
        tag="Tag"
        media={<x.img src="http://placekitten.com/500/300" alt="cat" />}
        body={<LoremIpsum units="sentences" />}
      />

      <Card
        tag="Tag"
        media={<x.img src="http://placekitten.com/500/300" alt="cat" />}
        body={<LoremIpsum units="sentences" />}
      />

      <Card
        title="Thick"
        tag="Tag"
        media={<x.img src="http://placekitten.com/500/300" alt="cat" />}
        body={<LoremIpsum units="sentences" />}
      />
    </>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

const LoremIpsum: FC<any> = () => {
  return (
    <>text text text</>
  );
};
