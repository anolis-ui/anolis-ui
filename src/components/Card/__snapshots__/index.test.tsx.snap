// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component 1`] = `
Array [
  .c0.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: text;
  background: white;
  max-width: 30rem;
}

.c1.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  padding: 1rem;
}

.c3.c3 {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4.c4 {
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
}

.c2.c2 {
  font-size: 2rem;
  line-height: 3rem;
}

<div
    className="c0"
    role="group"
  >
    <div
      className="c1"
    >
      <span
        className="c2"
      >
        Card title
      </span>
    </div>
    <div
      className="c3"
    >
      <img
        alt="cat"
        className=""
        src="http://placekitten.com/500/300"
      />
    </div>
    <div
      className="c4"
    >
      text text text
    </div>
    <div
      className="c4"
    >
      <p
        className=""
      >
        Footer
      </p>
    </div>
  </div>,
  .c0.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: text;
  background: white;
  max-width: 30rem;
}

.c1.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  padding: 1rem;
}

.c3.c3 {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4.c4 {
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
}

.c2.c2 {
  font-size: 2rem;
  line-height: 3rem;
}

<div
    className="c0"
    role="group"
  >
    <div
      className="c1"
    >
      <span
        className="c2"
      >
        Card title
      </span>
    </div>
    <div
      className="c3"
    >
      <img
        alt="cat"
        className=""
        src="http://placekitten.com/500/300"
      />
    </div>
    <div
      className="c4"
    >
      text text text
    </div>
    <div
      className="c4"
    >
      <p
        className=""
      >
        Footer
      </p>
    </div>
  </div>,
  .c0.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: text;
  background: white;
  max-width: 30rem;
}

.c1.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  padding: 1rem;
}

.c4.c4 {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5.c5 {
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
}

.c3.c3 {
  white-space: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: color 300ms,background 300ms,border-color 300ms;
  transition: color 300ms,background 300ms,border-color 300ms;
  border-radius: 0.3rem;
  border: 1px solid #455663;
  padding-right: 0.8rem;
  padding-left: 0.8rem;
  font-size: 1.2rem;
  font-weight: 500;
  line-height: 2rem;
  background-color: #455663;
  color: white;
  position: absolute;
  top: 0;
  right: 0;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c2.c2 {
  font-size: 2rem;
  line-height: 3rem;
}

<div
    className="c0"
    role="group"
  >
    <div
      className="c1"
    >
      <span
        className="c2"
      >
        Card title
      </span>
    </div>
    <div
      className="c3"
    >
      Tag
    </div>
    <div
      className="c4"
    >
      <img
        alt="cat"
        className=""
        src="http://placekitten.com/500/300"
      />
    </div>
    <div
      className="c5"
    >
      text text text
    </div>
  </div>,
  .c0.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: text;
  background: white;
  max-width: 30rem;
}

.c1.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  padding: 1rem;
}

.c4.c4 {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5.c5 {
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
}

.c3.c3 {
  white-space: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: color 300ms,background 300ms,border-color 300ms;
  transition: color 300ms,background 300ms,border-color 300ms;
  border-radius: 0.3rem;
  border: 1px solid #455663;
  padding-right: 0.8rem;
  padding-left: 0.8rem;
  font-size: 1.2rem;
  font-weight: 500;
  line-height: 2rem;
  background-color: #455663;
  color: white;
  position: absolute;
  top: 0;
  right: 0;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c2.c2 {
  font-size: 2rem;
  line-height: 3rem;
}

<div
    className="c0"
    role="group"
  >
    <div
      className="c1"
    >
      <span
        className="c2"
      >
        Card title
      </span>
    </div>
    <div
      className="c3"
    >
      Tag
    </div>
    <div
      className="c4"
    >
      <img
        alt="cat"
        className=""
        src="http://placekitten.com/500/300"
      />
    </div>
    <div
      className="c5"
    >
      text text text
    </div>
  </div>,
  .c0.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: text;
  background: white;
  max-width: 30rem;
}

.c2.c2 {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3.c3 {
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
}

.c1.c1 {
  white-space: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: color 300ms,background 300ms,border-color 300ms;
  transition: color 300ms,background 300ms,border-color 300ms;
  border-radius: 0.3rem;
  border: 1px solid #455663;
  padding-right: 0.8rem;
  padding-left: 0.8rem;
  font-size: 1.2rem;
  font-weight: 500;
  line-height: 2rem;
  background-color: #455663;
  color: white;
  position: absolute;
  top: 0;
  right: 0;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

<div
    className="c0"
    role="group"
  >
    <div
      className="c1"
    >
      Tag
    </div>
    <div
      className="c2"
    >
      <img
        alt="cat"
        className=""
        src="http://placekitten.com/500/300"
      />
    </div>
    <div
      className="c3"
    >
      text text text
    </div>
  </div>,
  .c0.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: text;
  background: white;
  max-width: 30rem;
}

.c1.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  padding: 1rem;
}

.c4.c4 {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5.c5 {
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
}

.c3.c3 {
  white-space: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: color 300ms,background 300ms,border-color 300ms;
  transition: color 300ms,background 300ms,border-color 300ms;
  border-radius: 0.3rem;
  border: 1px solid #455663;
  padding-right: 0.8rem;
  padding-left: 0.8rem;
  font-size: 1.2rem;
  font-weight: 500;
  line-height: 2rem;
  background-color: #455663;
  color: white;
  position: absolute;
  top: 0;
  right: 0;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c2.c2 {
  font-size: 2rem;
  line-height: 3rem;
}

<div
    className="c0"
    role="group"
  >
    <div
      className="c1"
    >
      <span
        className="c2"
      >
        Thick
      </span>
    </div>
    <div
      className="c3"
    >
      Tag
    </div>
    <div
      className="c4"
    >
      <img
        alt="cat"
        className=""
        src="http://placekitten.com/500/300"
      />
    </div>
    <div
      className="c5"
    >
      text text text
    </div>
  </div>,
]
`;
